daemon/anonymous-statistics.sh

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env sh
+
+# Valid actions:
+
+# - FATAL       - netdata exited due to a fatal condition
+#      ACTION_RESULT  -- program name and thread tag
+#      ACTION_DATA    -- fmt, args passed to fatal
+# - START       - netdata started
+#      ACTION_DATA     -- nan
+# - EXIT     - installation action
+#      ACTION_DATA    -- ret value of
+
+
+ACTION="${1}"
+ACTION_RESULT="${2}"
+ACTION_DATA="${3}"
+
+# -------------------------------------------------------------------------------------------------
+# check opt-out
+
+if [ -z "${NETDATA_CONFIG_DIR}" ]; then
+    for dir in "$(netdata -W get global 'config directory' '/etc/netdata' 2>/dev/null)" "/etc/netdata" "/opt/netdata/etc/netdata"; do
+        if [ -f "${dir}/.opt-out-from-anonymous-statistics" ]; then
+            exit 0
+        fi
+    done
+else
+    if [ -f "${NETDATA_CONFIG_DIR}/.opt-out-from-anonymous-statistics" ]; then
+        exit 0
+    fi
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the operating system
+
+OS_DETECTION="unknown"
+NAME="unknown"
+VERSION="unknown"
+VERSION_ID="unknown"
+ID="unknown"
+ID_LIKE="unknown"
+
+if [ -f "/etc/os-release" ]; then
+    OS_DETECTION="/etc/os-release"
+    eval "$(grep -E "^(NAME|ID|ID_LIKE|VERSION|VERSION_ID)=" </etc/os-release)"
+elif [ -f "/etc/lsb-release" ]; then
+    OS_DETECTION="/etc/lsb-release"
+    DISTRIB_ID="unknown"
+    DISTRIB_RELEASE="unknown"
+    DISTRIB_CODENAME="unknown"
+    eval "$(grep -E "^(DISTRIB_ID|DISTRIB_RELEASE|DISTRIB_CODENAME)=" </etc/lsb-release)"
+    NAME="${DISTRIB_ID}"
+    VERSION="${DISTRIB_RELEASE}"
+    ID="${DISTRIB_CODENAME}"
+elif [ ! -z "$(command -v lsb_release 2>/dev/null)" ]; then
+    OS_DETECTION="lsb_release"
+    NAME="$(lsb_release -i 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
+    VERSION_ID="$(lsb_release -f 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
```

Comment:
`lsb_release` doesn't support `-f` parameter.

```sh
root@4036e62f0da6:/# lsb_release --help
Usage: lsb_release [options]

Options:
  -h, --help         show this help message and exit
  -v, --version      show LSB modules this system supports
  -i, --id           show distributor ID
  -d, --description  show description of this distribution
  -r, --release      show release number of this distribution
  -c, --codename     show code name of this distribution
  -a, --all          show all of the above information
  -s, --short        show requested information in short format
```

Reply:
Also there is no need for all this parsing since `lsb_release` supports `-s` option which simplifies lines like this:
```
NAME="$(lsb_release -i 2>/dev/null | sed "s/[[:space:]]*:[[:space:]]*/:/g" | cut -d ':' -f 2)"
```

to 
```
NAME="$(lsb_release -is 2>/dev/null)"
```

Reply:
-d was wrong, I used -r in the proper place.